src/Common/src/System/Net/CookieParser.cs

```diff
@@ -662,10 +662,9 @@ internal Cookie Get()
                                     {
                                         expiresSet = true;
 
-                                        if (DateTime.TryParse(CheckQuoted(_tokenizer.Value),
-                                            CultureInfo.InvariantCulture, DateTimeStyles.AllowWhiteSpaces, out DateTime expires))
+                                        if (HttpDateParser.TryStringToDate(CheckQuoted(_tokenizer.Value), out DateTimeOffset expires))
                                         {
-                                            cookie.Expires = expires;
+                                            cookie.Expires = expires.UtcDateTime;
```

Comment:
Can we use `expires.DateTime` here?

Reply:
DateTimeOffset.DateTime returns a DateTime with a Kind set to Unspecified which is why I felt it would be better to use UtcDateTime

Reply:
Hmmm...this brings up an interesting issue.  We shouldn't change the semantics of this. So, we should use whatever time zone it was designed for.  

What does the code/public API design pattern currently use/assume?  We need to preserve that.

Reply:
>DateTimeOffset.DateTime returns a DateTime with a Kind set to Unspecified which is why I felt it would be better to use UtcDateTime

Why is it "Unspecified" in the first place?  The original information came from a string that specified a timezone such as GMT etc.

Reply:
`DatetimeOffset` doesn't retain information about the source string.  It only knows its offset, and just because the offset is zero doesn't necessarily mean it originated from UTC.  (For example, it may be the local time in London or Ireland in the winter months - when daylight saving time is not in effect.)  `DateTimeOffset.UtcDateTime` is indeed appropriate here (IMHO).

Reply:
I'm not sure why the Common/src HttpDateParser class returns the time as `DateTimeOffset` in the first place.  The input string is usually coming from the wire during an HTTP request/response and would have a timezone (normally GMT).

@stephentoub Do you know why the `HttpDateParser` class returns DateTimeOffset as the type?

Reply:
> Do you know why the HttpDateParser class returns DateTimeOffset as the type?

Don't know.

Reply:
Looks like this class was introduced in #22843 by @Clockwork-Muse.  It seemed to inherit the DateTimeOffset type from when this class was internally named as "HttpRuleParser".

@Clockwork-Muse can you comment on this?

At this point, it's probably not safe to consider changing the Common HttpDateParser class. But since this is parsing a string that presumably is required to have GMT timezone in it, the changes in this PR to allow "UTC" (which means the same as "GMT" probably means we should set the timezone to UTC.   So, using `expires.UtcDateTime` is probably appropriate.

Reply:
@davidsh - 
Yes, that's why it's `DateTimeOffset`, because it came from `HttpRuleParser`, to keep the type at the use site, and, likes @mj1856 says, since other offsets were possible.  As a concept, that's what's being passed, too: an offset date-time.
Note that trying to reuse `HttpDateParser` is going to really widen the allowed strings, although as a practical matter I doubt we're going to run into much


Speaking pedantically/about concepts, `DateTimeOffset` is still the wrong type, even if you could attach a timezone to it: the correct type is really more [`Instant`](https://nodatime.org/2.3.x/api/NodaTime.Instant.html), which isn't actually in the _timezone_ UTC (or any other).  I'm assuming the formats used are because HTTP 1- designers wanted something locally readable, which is probably why the formats are non-SARGable and contain day-of-week.


UTC, despite how much emphasis most programmers put on it, is strangely rarely relevant - rather, what's relevant is a business-chosen timezone (which might end up being UTC, because it has useful properties)